crates/node/src/scheduler/mod.rs

```diff
@@ -220,10 +228,19 @@ impl Scheduler {
                             sleep(Duration::from_millis(500)).await;
 
                             // Return the popped program_id back to pending queue.
-                            state.pending_programs.push_front((tx_hash, program_id));
+                            state.pending_programs.push_back((tx_hash, program_id));
```

Comment:
I feel like this old code here would have caused infinite loop, right? :grimacing: